portal-api/src/v1/users/routes.js

```diff
@@ -154,32 +155,44 @@ module.exports.findById = (req, res, next) => {
 };
 
 module.exports.updateById = (req, res, next) => {
-  findOne(req.params._id, (error, user) => {
-    if (error) {
-      next(error);
-    } else if (user) {
+  try {
+    const userIsAdmin = req.user?.roles?.includes(ADMIN);
+    const userIsChangingTheirOwnPassword = req.user?._id?.toString() === req.params._id
+    && !Object.keys(req.body).some(
+      (property) => !['password', 'passwordConfirm', 'currentPassword'].includes(property)
+    );
+    if (!userIsAdmin && !userIsChangingTheirOwnPassword) {
+      return res.status(403).send();
+    }
+    return findOne(req.params._id, (error, user) => {
+      if (error) {
+        return next(error);
+      }
+      if (!user) {
+        return res.status(404).send();
```

Comment:
Could we add some logging here of `console.error(`Error updating user: could not find user for id :${req.params.id}`)` or something here? Sorry I'm just getting you to improve existing code that aren't your changes :D